src/Test/ArchitectureMarkupTest.php

```diff
@@ -46,9 +46,12 @@ class ArchitectureMarkupTest implements TestInterface
      */
     private function invokeTest(string $method): Rule
     {
-        /** @var Rule $rule */
         $rule = call_user_func($this->$method);
 
+        if (!($rule instanceof Rule)) {
+            throw new \Exception($method . ' must return an instance of ' . Rule::class . '.');
+        }
```

Comment:
The way the error conditions are handled right now is via FatalErrorEvent (which already cuts the execution). Any other Exception thrown in the middle will result into this awful message https://github.com/carlosas/phpat/blob/b331edf31dd891295ed2b43d2c9a288d6fa00db1/phpat#L53

We can discuss in the _Symfony Devs_'s phpat channel if this is a good or bad approach but for now it is what it is. This change will need a refactor on how errors are handled all over the library